client/lib/api/notifications.dart

```diff
@@ -97,16 +99,16 @@ class Notifications {
     final tokenToSet =
         notificationsEnabled ? await _firebaseMessaging.getToken() : null;
 
-    final storedToken = await _userPrefs.getFirebaseToken();
-    if (tokenToSet != storedToken) {
-      await setFirebaseToken(tokenToSet);
-    }
+    await setFirebaseToken(tokenToSet);
   }
 
   Future setFirebaseToken(String newToken) async {
-    var exitsingToken = await _userPrefs.getFirebaseToken();
-    if (exitsingToken != newToken) {
-      await service.putDeviceToken(newToken);
+    var existingToken = await _userPrefs.getFirebaseToken();
+    var countryCode = await prefs.countryIsoCode;
+    // If countryCode has not yet been set, no need to update the server.
+    if (existingToken != newToken && countryCode != null) {
```

Comment:
Shouldn't this update the token even if the countryCode is null? If the `countryCode` is null mistakenly, I'd rather see that as an error server side rather than hidden on the client.

If anything is silently failing, better to add a `analytics.logEvent` to capture it.

Reply:
Yeah this comment may be wrong. I think it's probably better to have it not trigger until onboarding is complete?
I'm trying to avoid having two requests in flight if the token and countrycode 'show up' at the same time. That's one of the two possible race conditions in the current system, since these two events are async. That's why I have lots of open questions in the doc I shared!